src/common/copilot/PowerPagesCopilot.ts

```diff
@@ -45,13 +111,187 @@ export class PowerPagesCopilot implements vscode.WebviewViewProvider {
+
+          vscode.env.clipboard.writeText(data.value);
+          vscode.window.showInformationMessage(vscode.l10n.t('Copied to clipboard!'))
+          sendTelemetryEvent(this.telemetry, { eventName: CopyCodeToClipboardEvent, copilotSessionId: sessionID });
+          break;
+        }
+        case "clearChat": {
+
+          sessionID = uuidv4();
+          break;
+        }
+        case "userFeedback": {
 
+          if (data.value === "thumbsUp") {
+
+            sendTelemetryEvent(this.telemetry, { eventName: UserFeedbackThumbsUpEvent, copilotSessionId: sessionID });
+            CESUserFeedback(this._extensionContext, sessionID, userID, "thumbsUp", this.telemetry)
+          } else if (data.value === "thumbsDown") {
+
+            sendTelemetryEvent(this.telemetry, { eventName: UserFeedbackThumbsDownEvent, copilotSessionId: sessionID });
+            CESUserFeedback(this._extensionContext, sessionID, userID, "thumbsDown", this.telemetry)
+          }
+        }
+      }
+    });
+  }
+
+  public show() {
+    if (this._view) {
+      // Show the webview view
+      this._view.show(true);
+    }
   }
 
+  private async handleLogin() {
+
+    const pacOutput = await this._pacWrapper.activeOrg();
+    if (pacOutput.Status === "Success") {
+      this.handleOrgChangeSuccess.call(this, pacOutput);
+
+      intelligenceAPIAuthentication().then(({ accessToken, user }) => {
+        this.intelligenceAPIAuthenticationHandler.call(this, accessToken, user);
+      });
+
+    } else if (this._view?.visible) {
+
+      const userOrgUrl = await showInputBoxAndGetOrgUrl();
+      if (!userOrgUrl) {
+        userName = "";
+        this.sendMessageToWebview({ type: 'userName', value: userName });
+
+        if (!this.loginButtonRendered) {
+          this.sendMessageToWebview({ type: "welcomeScreen" });
+          this.loginButtonRendered = true; // Set the flag to indicate that the login button has been rendered
+        }
+
+        return;
+      }
+      const pacAuthCreateOutput = await this._pacWrapper.authCreateNewAuthProfileForOrg(userOrgUrl);
+      pacAuthCreateOutput.Status === "Success"
+      ? intelligenceAPIAuthentication().then(({ accessToken, user }) =>
+          this.intelligenceAPIAuthenticationHandler.call(this, accessToken, user)
+        )
+      : vscode.window.showErrorMessage("Error creating auth profile for org");
+    
+
+    }
+  }
+
+  private authenticateAndSendAPIRequest(data: string, activeFileParams: ActiveFileParams, orgID: string) {
+    return intelligenceAPIAuthentication()
+      .then(({ accessToken, user }) => {
+        apiToken = accessToken;
+        userName = getUserName(user);
+        this.sendMessageToWebview({ type: 'userName', value: userName });
+
+        return sendApiRequest(data, activeFileParams, orgID, apiToken);
+      })
+      .then(apiResponse => {
+        this.sendMessageToWebview({ type: 'apiResponse', value: apiResponse });
+        this.sendMessageToWebview({ type: 'enableInput' });
```

Comment:
what happens to error scenarios

Reply:
That is handled in API service. It passes the response to webview for user.